transaction/core/src/ring_signature/mod.rs

```diff
@@ -33,9 +33,9 @@ lazy_static! {
 
     /// Generators (base points) for Bulletproofs.
     /// The `party_capacity` is the maximum number of values in one proof. It should
-    /// be at least MAX_OUTPUTS to allow for all outputs in a transaction
+    /// be at least 2 * MAX_INPUTS + MAX_OUTPUTS, which allows for inputs, pseudo outputs, and outputs.
```

Comment:
Orthogonal to our immediate need, but were we ever including range proofs for the inputs?

Reply:
Great question. Inputs are assumed to be in the valid because they are already in the ledger. If that is the "inductive step", the "base case" is that all outputs created by the origin block are valid. We prove that by publishing the view key for the origin block.